src/ImageSharp/Formats/Webp/Lossy/LossyUtils.cs

```diff
@@ -1550,6 +2054,203 @@ private static bool NeedsFilter2(Span<byte> p, int offset, int step, int t, int
                    WebpLookupTables.Abs0(q2 - q1) <= it && WebpLookupTables.Abs0(q1 - q0) <= it;
         }
 
+#if SUPPORTS_RUNTIME_INTRINSICS
+        private static Vector128<byte> NeedsFilter(Vector128<byte> p1, Vector128<byte> p0, Vector128<byte> q0, Vector128<byte> q1, int thresh)
+        {
+            var mthresh = Vector128.Create((byte)thresh);
+            Vector128<byte> t1 = Abs(p1, q1); // abs(p1 - q1)
+            var fe = Vector128.Create((byte)0xFE);
+            Vector128<byte> t2 = Sse2.And(t1, fe); // set lsb of each byte to zero.
+            Vector128<short> t3 = Sse2.ShiftRightLogical(t2.AsInt16(), 1); // abs(p1 - q1) / 2
+
+            Vector128<byte> t4 = Abs(p0, q0); // abs(p0 - q0)
+            Vector128<byte> t5 = Sse2.AddSaturate(t4, t4); // abs(p0 - q0) * 2
+            Vector128<byte> t6 = Sse2.AddSaturate(t5.AsByte(), t3.AsByte()); // abs(p0-q0)*2 + abs(p1-q1)/2
+
+            Vector128<byte> t7 = Sse2.SubtractSaturate(t6, mthresh.AsByte()); // mask <= m_thresh
+
+            return Sse2.CompareEqual(t7, Vector128<byte>.Zero);
+        }
+
+        private static void Load16x4(Span<byte> r0, Span<byte> r8, int stride, out Vector128<byte> p1, out Vector128<byte> p0, out Vector128<byte> q0, out Vector128<byte> q1)
```

Comment:
Can this and `Load8x4` accept ref bytes? You might be able to avoid some of the slicing then (_assuming this operation is performed in a loop and it's advantageous to avoid_)

Reply:
Changed with 61078454